apps/core/lib/core/schema/user.ex

```diff
@@ -159,6 +160,19 @@ defmodule Core.Schema.User do
     |> generate_uuid(:avatar_id)
     |> change_markers(password_hash: :password_change)
     |> cast_attachments(attrs, [:avatar], allow_urls: true)
+    |> set_email_changed(model)
+  end
+
+  def set_email_changed(cs, model) do
+    confirm_by = Timex.now() |> Timex.shift(days: 7)
+
+    case get_change(cs, :email) do
+      email when is_binary(email) and email != model.email ->
```

Comment:
are you sure you need to do the `email != model.email` check?

Reply:
@michaeljguarino You are right, it's redundant as `get_change` will return `nil` if email has not changed. Then it will not pass `is_binary` check. Fixed, PTAL.